infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/CertificateProvisioner.java

```diff
@@ -83,15 +84,17 @@ public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity identity)
                 .build());
 
     for (PodData pod : k8sEnv.getPodsData().values()) {
-      Optional<Volume> certVolume =
-          pod.getSpec()
-              .getVolumes()
-              .stream()
-              .filter(v -> v.getName().equals(CHE_SELF_SIGNED_CERT_VOLUME))
-              .findAny();
-
-      if (!certVolume.isPresent()) {
-        pod.getSpec().getVolumes().add(buildCertSecretVolume(selfSignedCertSecretName));
+      if (pod.getRole() == PodRole.DEPLOYMENT) {
```

Comment:
I think this is equivalent, but shorter and simpler 
```
        if (pod.getSpec().getVolumes().stream()
            .noneMatch(v -> v.getName().equals(CHE_SELF_SIGNED_CERT_VOLUME))) {
          pod.getSpec().getVolumes().add(buildCertSecretVolume(selfSignedCertSecretName));
        }
```

Reply:
Yeah, I just wrapped existing code in an `if` in trying to keep the amount of change to the minimum. But you're right, this can be simplified.